cartography/intel/aws/ssm.py

```diff
@@ -0,0 +1,193 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import boto3
+import neo4j
+
+from cartography.util import aws_handle_regions
+from cartography.util import dict_date_to_epoch
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_instance_ids(neo4j_session: neo4j.Session, region: str, current_aws_account_id: str) -> List[str]:
+    get_instances_query = """
+    MATCH (:AWSAccount{id: {AWS_ACCOUNT_ID}})-[:RESOURCE]->(i:EC2Instance)
+    WHERE i.region = {Region}
+    RETURN i.id
+    """
+    results = neo4j_session.run(get_instances_query, AWS_ACCOUNT_ID=current_aws_account_id, Region=region)
+    instance_ids = []
+    for r in results:
+        instance_ids.append(r['i.id'])
+    return instance_ids
+
+
+@timeit
+@aws_handle_regions
+def get_instance_information(
+    boto3_session: boto3.session.Session, region: str, instance_ids: List[str],
+) -> List[Dict[str, Any]]:
+    client = boto3_session.client('ssm', region_name=region)
+    instance_information: List[Dict[str, Any]] = []
+    paginator = client.get_paginator('describe_instance_information')
+    for i in range(0, len(instance_ids), 50):
+        instance_ids_chunk = instance_ids[i:i + 50]
+        for info_chunk in paginator.paginate(
+            Filters=[{"Key": "InstanceIds", "Values": instance_ids_chunk}],
+            MaxResults=50,
+        ):
+            instance_information.extend(info_chunk.get('InstanceInformationList', []))
+    return instance_information
+
+
+@timeit
+@aws_handle_regions
+def get_instance_patches(
+    boto3_session: boto3.session.Session, region: str, instance_ids: List[str],
+) -> List[Dict[str, Any]]:
+    client = boto3_session.client('ssm', region_name=region)
+    instance_patches: List[Dict[str, Any]] = []
+    paginator = client.get_paginator('describe_instance_patches')
+    for instance_id in instance_ids:
+        patches = []
+        for page in paginator.paginate(InstanceId=instance_id):
+            patches.extend(page["Patches"])
+        # to avoid complicating the load function, inject the instance ID into the patch
+        for patch in patches:
+            patch["_instance_id"] = instance_id
+        instance_patches.extend(patches)
+    return instance_patches
+
+
+@timeit
+def load_instance_information(
+    neo4j_session: neo4j.Session,
+    data: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    ingest_query = """
+    UNWIND {InstanceInformation} AS instance
+        MERGE (i:SSMInstanceInformation{id: instance.InstanceId})
```

Comment:
non-blocker I think `activation_id` would be a better candidate for the node's id.
https://lyft.github.io/cartography/dev/writing-intel-modules.html#all-nodes-need-these-fields

Reply:
`activation_id` isn't the unique ID for the record, as activations are another feature of SSM, and if it isn't enabled, that field is null. `instance_id` is unique and always set, as the instance information is one to one with instances.